wordpress-mobile/WordPress-FluxC-Android
# Add support for connecting to self-hosted sites without XMLRPC

Part of #2648 
Based on the work done in the previous PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2583

The PR adds a new way for fetching sites directly using the [Cookie authentication](https://developer.wordpress.org/rest-api/using-the-rest-api/authentication/), the changes of the PR are:

- Add a new Origin value to the SiteModel.
- Create a new rest client SiteWPAPIRestClient that creates a SiteModel instance using the /wp-json response, and uses the Cookie authentication.
- Integrates the new rest client to SiteStore and add a new action to fetch a site using Cookie Authentication.
- Update the example app to test the new flow.
- Update the WPAPI requests to emit authentication errors when they are encountered.

### Test
1. Open the example app.
2. Click on "Sign in and fetch sites"
3. Enter your wp-admin credentials and the site URL, and make sure the Checkbox is kept unchecked.
4. Click on OK.
5. Confirm that the site name is correctly printed in the console.
6. Click on Plugins.
7. Test fetching/installing some plugins.
8. Confirm it works as expected.


Bonus: you can also use a plugin (for example this [one](https://wordpress.org/plugins/disable-xml-rpc-api/)) to disable XMLPRC to confirm that these changes work without it.
